s3/src/main/scala/akka/stream/alpakka/s3/impl/SplitAfterSizeWithContext.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.s3.impl
+
+import akka.annotation.InternalApi
+import akka.stream.scaladsl.{Flow, SubFlow}
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler}
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+import akka.util.ByteString
+import com.github.ghik.silencer.silent
+
+/**
+ * Internal Api
+ *
+ * Splits up a byte stream source into sub-flows of a minimum size while maintaining a Context. Does not attempt to
+ * create chunks of an exact size. Unlike `SplitAfterSize` there is no maximum size since this would imply splitting
+ * up a Context which is not possible to do if the Context is generic.
+ *
+ * This also means that `SplitAfterSizeContext` currently doesn't support buffering since that would require a way to
+ * serialize the Context in the case of a disk buffer which is currently unsupported (we would have to add in a
+ * C => ByteString serializer in the public API)
+ *
+ */
+@InternalApi private[impl] object SplitAfterSizeWithContext {
+  // The warning being suppressed is abstract type pattern C is unchecked since it is eliminated by erasure which
+  // happens in the pattern match on collect.
+  @silent
```

Comment:
can it be `case (bs: ByteString, context: C @unchecked)` instead of silencing the warning?

Reply:
Ah so that is what `@unchecked` does, I had no idea, doing this now!